tidy3d/components/data/monitor_data.py

```diff
@@ -174,7 +204,8 @@ class ElectromagneticFieldData(AbstractFieldData, ElectromagneticFieldDataset, A
     def _expanded_grid_field_coords(self, field_name: str):
         """Coordinates in the expanded grid corresponding to a given field component."""
         if self.monitor.colocate:
-            return self.grid_expanded["centers"]
+            bounds_dict = self.grid_expanded.boundaries.to_dict
+            return Coords(**{key: val[:-1] for key, val in bounds_dict.items()})
```

Comment:
There are as many fields as there are *cells* in the grid, while the number of grid boundaries is one more than the number of cells. In other words, if the grid is `|_|_|` and monitor.colocate is True, all fields live on the first two boundaries.

Now, thinking about it I realize that it may be nice to just have fields on all three boundaries in that case, but the current way makes things a bit more nicely unified with the `colocate=False` case (where tangential fields live on the first two boundaries, while normal fields live on the first two centers, i.e. they are just recorded in the first two pixels), and is in general kinda assumed in the backend codebase, so I wouldn't change it unless there's a pretty good reason.